### ISSUE TITLE
Add support for arbitrarily splitting log file to log entries

### ISSUE THREAD

**Issue by modernist** [RolandPheasant/TailBlazer#165]
Most log files have some kind of prologue in each entry (e.g. the timestamp of an event), but each entry can span multiple lines.

It would be useful to allow the user to define a regex that will be used to split the log file into entries and handle each entry as a single "line" regardless of the actual number of lines it may span.

**Comment by RolandPheasant** [comment]
I agree, it would be a good feature. When I clear my current backlog I will take a look
